description: fix undefined order in which expressions are evaluated (-Wsequence-point)

now upstream:

From a00fc336667973490b965e769b2aaeb53c4659d0 Mon Sep 17 00:00:00 2001
From: Steinar H. Gunderson <sesse@chromium.org>
Date: Wed, 19 Jan 2022 10:40:41 +0000
Subject: [PATCH] Fix undefined order in which expressions are evaluated (-Wsequence-point).

This is a patch taken from the Debian Chromium package, originally
by Michael Gilbert <mgilbert@debian.org> (but with some minor
adjustments to make it apply to head).

Original patch:
https://salsa.debian.org/chromium-team/chromium/-/blob/master/debian/patches/fixes/sequence-point.patch

Change-Id: I4115420e5ee49b76fd8080402c689c39961b5d8c
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3385318
Reviewed-by: David Trainor <dtrainor@chromium.org>
Reviewed-by: Robert Flack <flackr@chromium.org>
Commit-Queue: Steinar H Gunderson <sesse@chromium.org>
Cr-Commit-Position: refs/heads/main@{#960877}
---

diff --git a/chrome/browser/notifications/scheduler/internal/impression_history_tracker.cc b/chrome/browser/notifications/scheduler/internal/impression_history_tracker.cc
index fe52aa5..b59c754 100644
--- a/chrome/browser/notifications/scheduler/internal/impression_history_tracker.cc
+++ b/chrome/browser/notifications/scheduler/internal/impression_history_tracker.cc
@@ -416,7 +416,7 @@
 
   // Increase |current_max_daily_show| by 1.
   client_state->current_max_daily_show =
-      base::clamp(++client_state->current_max_daily_show, 0,
+      base::clamp(client_state->current_max_daily_show + 1, 0,
                   config_.max_daily_shown_per_type);
 }
 
diff --git a/components/download/internal/common/download_item_impl.cc b/components/download/internal/common/download_item_impl.cc
index 19c8408..685b91b 100644
--- a/components/download/internal/common/download_item_impl.cc
+++ b/components/download/internal/common/download_item_impl.cc
@@ -686,7 +686,9 @@
     bytes_wasted_ = 0;
   }
 
-  auto_resume_count_ = user_resume ? 0 : ++auto_resume_count_;
+  ++auto_resume_count_;
+  if (user_resume)
+    auto_resume_count_ = 0;
   download_schedule_ = absl::nullopt;
   RecordDownloadLaterEvent(DownloadLaterEvent::kScheduleRemoved);
 }
diff --git a/third_party/blink/renderer/modules/animationworklet/animation_worklet_proxy_client.cc b/third_party/blink/renderer/modules/animationworklet/animation_worklet_proxy_client.cc
index 64c3360..8d3bf6f 100644
--- a/third_party/blink/renderer/modules/animationworklet/animation_worklet_proxy_client.cc
+++ b/third_party/blink/renderer/modules/animationworklet/animation_worklet_proxy_client.cc
@@ -188,7 +188,7 @@
   if (--next_global_scope_switch_countdown_ < 0) {
     int last_global_scope_index = current_global_scope_index_;
     current_global_scope_index_ =
-        (++current_global_scope_index_ % global_scopes_.size());
+        (current_global_scope_index_ + 1) % global_scopes_.size();
     global_scopes_[last_global_scope_index]->MigrateAnimatorsTo(
         global_scopes_[current_global_scope_index_]);
     // Introduce an element of randomness in the switching interval to make
